pkg/model/openstackmodel/servergroup.go

```diff
@@ -207,6 +212,16 @@ func (b *ServerGroupModelBuilder) Build(c *fi.ModelBuilderContext) error {
 		if ig.Spec.Role == kops.InstanceGroupRoleMaster {
 			masters = append(masters, sgTask)
 		}
+
+		for _, secGroup := range ig.Spec.AdditionalSecurityGroups {
+			if _, ok := additionalSecurityGroups[secGroup]; !ok {
+				additionalSecurityGroups[secGroup] = b.LinkToSecurityGroup(secGroup)
+			}
+		}
+	}
+
+	for _, secGroup := range additionalSecurityGroups {
+		c.AddTask(secGroup)
```

Comment:
IMO we should not manage additional sec groups with kops. Or are you creating new security groups using kops?

Reply:
Not sure what you mean. What I saw is that the additional SG is not going to be deleted we the cluster gets deleted, although we have a task here for it. So I am not sure if it is safe to have a task for it or not. In the end we have to propagate it somehow to the port. But we could also "sneak" it directly there and add it only as a sort of metadata, which would work similar to the SGs for instances.

Reply:
well in my opinion this is not needed, its enough if you just "sneak it in"